consensus/ising/proposerservice.go

```diff
@@ -133,22 +129,9 @@ func (ps *ProposerService) ConsensusRoutine(vType voting.VotingContentType) {
 		// process final block and signature chain
 		switch vType {
 		case voting.SigChainTxnVote:
-			txn := content.(*transaction.Transaction)
-			payload := txn.Payload.(*payload.Commit)
-			sigchain := &por.SigChain{}
-			proto.Unmarshal(payload.SigChain, sigchain)
-			// TODO: get a determinate public key on signature chain
-			pbk, err := sigchain.GetLedgerNodePubkey()
-			if err != nil {
-				log.Warn("Get last public key error", err)
-				return
+			if txn, ok := content.(*transaction.Transaction); ok {
```

Comment:
"*transaction.Transaction" ?

Reply:
Transaction structure in transaction package. We could make this more logical when re-architect.